tests/integration/api/detection/test_detection.py

```diff
@@ -0,0 +1,296 @@
+"""API Tests for detection training"""
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import glob
+import os.path as osp
+import random
+import time
+import warnings
+from concurrent.futures import ThreadPoolExecutor
+from typing import Optional
+
+import numpy as np
+
+from otx.algorithms.common.tasks.training_base import BaseTask
+from otx.algorithms.detection.tasks import DetectionInferenceTask, DetectionTrainTask
+from otx.algorithms.detection.utils import generate_label_schema
+from otx.api.configuration.helper import create
+from otx.api.entities.annotation import AnnotationSceneEntity, AnnotationSceneKind
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.image import Image
+from otx.api.entities.inference_parameters import InferenceParameters
+from otx.api.entities.metrics import Performance
+from otx.api.entities.model import ModelEntity
+from otx.api.entities.model_template import (
+    TaskType,
+    parse_model_template,
+    task_type_to_label_domain,
+)
+from otx.api.entities.resultset import ResultSetEntity
+from otx.api.entities.subset import Subset
+from otx.api.entities.task_environment import TaskEnvironment
+from otx.api.entities.train_parameters import TrainParameters
+from otx.api.usecases.tasks.interfaces.export_interface import ExportType
+from otx.api.utils.shape_factory import ShapeFactory
+from tests.test_suite import e2e_pytest_api
+from tests.unit.api.test_helpers import generate_random_annotated_image
+
+DEFAULT_DET_TEMPLATE_DIR = osp.join("configs", "detection", "mobilenetv2_atss_cls_incr")
+
+
+def eval(task: BaseTask, model: ModelEntity, dataset: DatasetEntity) -> Performance:
+    start_time = time.time()
+    result_dataset = task.infer(dataset.with_empty_annotations())
+    end_time = time.time()
+    print(f"{len(dataset)} analysed in {end_time - start_time} seconds")
+    result_set = ResultSetEntity(
+        model=model, ground_truth_dataset=dataset, prediction_dataset=result_dataset
+    )
+    task.evaluate(result_set)
+    assert result_set.performance is not None
+    return result_set.performance
+
+
+class TestDetectionTaskAPI:
+    """
+    Collection of tests for OTE API and OTE Model Templates
+    """
+
+    @e2e_pytest_api
+    def test_reading_detection_cls_incr_model_template(self):
+        detection_template = ["mobilenetv2_atss_cls_incr"]
```

Comment:
Is atss the only supported model?

Reply:
I'll edit that file and tell you again. Thanks for the comment.

Reply:
Fixed!